### PR TITLE
Add XWayland

### PR DESCRIPTION
**Pull request by flyingP0tat0**
Depends on #1228.

X-Server with (only) Wayland support.

### CODE REVIEW

**Comment by jam7** [comment]
Missing dependency.
```
$ crew install libfontenc
...
checking if gcc -std=gnu99 supports -pedantic... yes
checking if gcc -std=gnu99 supports -Werror... yes
checking if gcc -std=gnu99 supports -Werror=attributes... yes
Package xorg-macros was not found in the pkg-config search path.
Perhaps you should add the directory containing `xorg-macros.pc'
to the PKG_CONFIG_PATH environment variable
No package 'xorg-macros' found
checking whether make supports nested variables... (cached) yes
checking for root directory for font files... Package fontutil was not found in the pkg-config search path.
Perhaps you should add the directory containing `fontutil.pc'
to the PKG_CONFIG_PATH environment variable
No package 'fontutil' found
${datarootdir}/fonts/X11
checking for directory for encodings files... ${FONTROOTDIR}/encodings
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking for gzclose in -lz... yes
checking for FONTENC... configure: error: Package requirements (xproto) were not met:

No package 'xproto' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables FONTENC_CFLAGS
and FONTENC_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

libfontenc failed to build: `./configure --prefix=/usr/local` exited with 1
```

**Comment by flyingP0tat0** [comment]
@jam7 Problem should be resolved.

**Comment by jam7** [comment]
Another errors unfortunately.
```
$ crew install libfontenc
...
checking if gcc -std=gnu99 supports -Werror=pointer-to-int-cast... yes
checking if gcc -std=gnu99 supports -pedantic... yes
checking if gcc -std=gnu99 supports -Werror... yes
checking if gcc -std=gnu99 supports -Werror=attributes... yes
Package xorg-macros was not found in the pkg-config search path.
Perhaps you should add the directory containing `xorg-macros.pc'
to the PKG_CONFIG_PATH environment variable
No package 'xorg-macros' found
checking whether make supports nested variables... (cached) yes
checking for root directory for font files... Package fontutil was not found in the pkg-config search path.
Perhaps you should add the directory containing `fontutil.pc'
to the PKG_CONFIG_PATH environment variable
No package 'fontutil' found
${datarootdir}/fonts/X11
checking for directory for encodings files... ${FONTROOTDIR}/encodings
checking zlib.h usability... no
checking zlib.h presence... no
checking for zlib.h... no
configure: error: in `/usr/local/tmp/crew/libfontenc-1.1.3.tar.gz.dir/libfontenc-1.1.3':
configure: error: zlib.h is required to compile libfontenc
See `config.log' for more details
libfontenc failed to build: `./configure --prefix=/usr/local` exited with 1
```

**Comment by flyingP0tat0** [comment]
@jam7 I added `zlibpkg`.

File: packages/xwayland.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  depends_on 'wayland'
+  depends_on 'eudev'
+  depends_on 'libxfont'
+  depends_on 'libbsd'
+  depends_on 'nettle'
+  depends_on 'libtirpc'
+  depends_on 'pixman'
+
+  def self.build
+    system "./configure --prefix=#{CREW_PREFIX} --disable-docs --disable-devel-docs --enable-xwayland --disable-xorg --disable-xvfb --disable-xnest --disable-xquartz --disable-xwin"
```
Comment (uberhacker) [inline packages/xwayland.rb]: Why are we disabling so many X configure options here?

Comment (flyingP0tat0) [reply packages/xwayland.rb]: Yes, they are not needed here, as we don't need the full X-Server, just the bare Wayland stuff to get Weston working.

Comment (uberhacker) [reply packages/xwayland.rb]: OK, cool.  Just curious.

**Comment by flyingP0tat0** [comment]
Any news on this one?

**Comment by uberhacker** [comment]
@flyingP0tat0: I can test it later this evening.  I'll let you know then.  Thanks

**Comment by uberhacker** [comment]
@flyingP0tat0: There are several issues that need to be corrected.
The libepoxy package installed no files.
```
$ crew files libepoxy
(i) libepoxy: Epoxy is a library for handling OpenGL function pointer management for you
https://github.com/anholt/libepoxy
version 1.4.3
Total found: 0
```
The libfontenc package needs to abstract the library files via `--libdir=#{CREW_LIB_PREFIX}`.
```
$ crew files libfontenc
(i) libfontenc: library for the X window system
https://x.org
version 1.1.3
/usr/local/include/X11/fonts/fontenc.h
/usr/local/lib/libfontenc.a
/usr/local/lib/libfontenc.la
/usr/local/lib/libfontenc.so
/usr/local/lib/libfontenc.so.1
/usr/local/lib/libfontenc.so.1.0.0
/usr/local/lib/pkgconfig/fontenc.pc
Total found: 7
```
The libxfont package needs to abstract the library files via `--libdir=#{CREW_LIB_PREFIX}`.
```
$ crew files libxfont
(i) libxfont: library for the X window system
https://x.org
version 2.0.1
/usr/local/include/X11/fonts/libxfont2.h
/usr/local/lib/libXfont2.a
/usr/local/lib/libXfont2.la
/usr/local/lib/libXfont2.so
/usr/local/lib/libXfont2.so.2
/usr/local/lib/libXfont2.so.2.0.0
/usr/local/lib/pkgconfig/xfont2.pc
Total found: 7
```
The libfontenc package needs to abstract the library files via `--libdir=#{CREW_LIB_PREFIX}`.
```
$ crew files libfontenc
(i) libfontenc: library for the X window system
https://x.org
version 1.1.3
/usr/local/include/X11/fonts/fontenc.h
/usr/local/lib/libfontenc.a
/usr/local/lib/libfontenc.la
/usr/local/lib/libfontenc.so
/usr/local/lib/libfontenc.so.1
/usr/local/lib/libfontenc.so.1.0.0
/usr/local/lib/pkgconfig/fontenc.pc
Total found: 7
```
The xorg_proto package did not install any files.
```
$ crew files xorg_proto
(i) xorg_proto
version 
Package xorg_proto is not installed. :(
```

**Comment by flyingP0tat0** [comment]
@uberhacker I corrected the prefix issues.

**Comment by uberhacker** [comment]
Thanks @flyingP0tat0.  Working nicely now.  @lyxell @thedamian @jam7, can you please check this also when you get a chance so we can have a second (or third) opinion if this is ready?

File: packages/xf86driproto.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+class Xf86driproto < Package
+  description 'The protocols for the X window system provide extended functionality for communication between a X client and the server.'
+  homepage 'https://x.org'
+  version '2.1.1'
+  source_url 'https://www.x.org/archive/individual/proto/xf86driproto-2.1.1.tar.gz'
+  source_sha256 '18ff8de129b89fa24a412a1ec1799f8687f96c186c655b44b1a714a3a5d15d6c'
+
+  def self.build
+    system "./configure --prefix=#{CREW_PREFIX}"
```
Comment (uberhacker) [inline packages/xf86driproto.rb]: I noticed there was one file in `/usr/local/lib`, so we probably need `--libdir=#{CREW_LIB_PREFIX}` in this package as well.  See below:
```
$ crew files xf86driproto
(i) xf86driproto: The protocols for the X window system provide extended functionality for communication between a X client and the server.
https://x.org
version 2.1.1
/usr/local/include/X11/dri/xf86dri.h
/usr/local/include/X11/dri/xf86driproto.h
/usr/local/include/X11/dri/xf86dristr.h
/usr/local/lib/pkgconfig/xf86driproto.pc
Total found: 4
```

File: packages/xwayland.rb (hunk 1 above)
Comment (uberhacker) [reply packages/xwayland.rb]: I noticed there was one file in `/usr/local/lib` so we probably need `--libdir=#{CREW_LIB_PREFIX}` in this package as well.  See below:
```
$ crew files xwayland
(i) xwayland: X server configured to (only) work with weston
https://x.org
version 1.19.3
/usr/local/bin/Xwayland
/usr/local/lib/xorg/protocol.txt
/usr/local/share/man/man1/Xserver.1
/usr/local/share/X11/xkb/compiled/README.compiled
Total found: 4
```

**Pull request merged**